datafusion/execution/src/registry.rs

```diff
@@ -44,10 +44,7 @@ pub trait FunctionRegistry {
     ///
     /// Returns an error (the default) if the function can not be registered,
     /// for example if the registry is read only.
-    fn register_udf(&mut self, _udf: Arc<ScalarUDF>) -> Result<Option<Arc<ScalarUDF>>> {
-        not_impl_err!("Registering ScalarUDF")
-    }
-
+    fn register_udf(&mut self, _udf: Arc<ScalarUDF>) -> Result<Option<Arc<ScalarUDF>>>;
```

Comment:
Why was the default implementation removed? 

If we remove the default implementation, it means that this change will no longer be backwards compatible because any code that  implements FunctionRegistry will now have to implement all the methods 

Thus, I think it would be better to leave the default (erroring) implementation and be backwards compatible

Reply:
OK, got it